### PR TITLE
Documentation and test improvements for NSX-T Network DHCP type

### PR DESCRIPTION
**Pull request by Didainius**
This PR:
* Adjusts `types.OpenApiOrgVdcNetworkDhcp` **documentation (only)** with latest DHCP features.
* Adjusts tests to verify DHCP with other types of network.

### CODE REVIEW

File: types/v56/nsxt_types.go [hunk 1]
```diff
@@ -201,28 +201,36 @@ type NsxtImportableSwitch = OpenApiReference
...
+	// for. The minimum is 60s while the maximum is 4,294,967,295s, which is roughly 49,710 days.
 	LeaseTime *int                            `json:"leaseTime,omitempty"`
 	DhcpPools []OpenApiOrgVdcNetworkDhcpPools `json:"dhcpPools,omitempty"`
+
 	// Mode describes how the DHCP service is configured for this network. Once a DHCP service has been created, the mode
 	// attribute cannot be changed. The mode field will default to 'EDGE' if it is not provided. This field only applies
 	// to networks backed by an NSX-T network provider.
 	//
-	// The supported values are EDGE (default) and NETWORK.
+	// The supported values are EDGE ,NETWORK and RELAY (VCD 10.3.1+, API 36.1+).
```
Comment (lvirbalas) [inline types/v56/nsxt_types.go:215]:
```suggestion
	// The supported values are EDGE, NETWORK and RELAY (VCD 10.3.1+, API 36.1+).
```

File: govcd/openapi_org_network_test.go [hunk 2]
```diff
@@ -435,10 +429,81 @@ func nsxtRoutedDhcpConfig(check *C, vcd *TestVCD, vdc *Vdc, orgNetId string) {
 	check.Assert(err, IsNil)
 	check.Assert(len(deletedDhcp.OpenApiOrgVdcNetworkDhcp.DhcpPools), Equals, 0)
 	check.Assert(len(deletedDhcp.OpenApiOrgVdcNetworkDhcp.DnsServers), Equals, 0)
+}
+
+// nsxtDhcpConfigNetworkMode checks DHCP functionality in NETWORK mode
```
Comment (lvirbalas) [inline govcd/openapi_org_network_test.go:434]:
```suggestion
// nsxtDhcpConfigNetworkMode checks DHCP functionality in NETWORK mode.
```

**Review by lvirbalas — APPROVED** [review]
Thanks! Only few nits.

File: types/v56/nsxt_types.go (hunk 1 above)
Comment (adambarreiro) [inline types/v56/nsxt_types.go:215]: nit
```suggestion
	// The supported values are EDGE, NETWORK and RELAY (VCD 10.3.1+, API 36.1+).
```

Comment (Didainius) [reply types/v56/nsxt_types.go:215]: Fixed.

File: govcd/openapi_org_network_test.go (hunk 2 above)
Comment (Didainius) [reply govcd/openapi_org_network_test.go:434]: fixed.

File: types/v56/nsxt_types.go (hunk 1 above)
Comment (Didainius) [reply types/v56/nsxt_types.go]: fixed.

**Review by adambarreiro — APPROVED** [review]

**Review by dataclouder — APPROVED** [review]

**Pull request merged**